src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -319,8 +319,9 @@ protected SCMRevision retrieve(SCMHead head, TaskListener listener)
                 // This was previously a GitTagSCMHead and needs to be property retrieved
                 // Perform a fetch of the tag from the remote.
                 // Create a new BitbucketSCMRevision from the fetched tag.
-                Optional<BitbucketTag> fetchedTag = fetchBitbucketTag((BitbucketTagSCMHead) head, listener);
-                return new BitbucketSCMRevision((BitbucketTagSCMHead) head, fetchedTag.map(BitbucketTag::getLatestCommit).orElse(null));
+                return fetchBitbucketCommit((BitbucketTagSCMHead) head).map(fetchedCommit -> {
+                    return new BitbucketSCMRevision((BitbucketTagSCMHead) head, fetchedCommit.getId());
```

Comment:
Is there a reason we're recycling the same `head` rather than constructing a new one from the fetched commits like the other methods? I think the case of the head being incorrect is much lower- you'd need to delete and recreate the tag for the persisted head to be wrong, but I've done that before :P We know the fetched commit is correct so that seems like the maximally safe thing to do here.